### PR TITLE
feat(TDKN-219) Add daikon-spring-metrics

### PR DESCRIPTION
**Pull request by fhuaulme**
**What is the problem this Pull Request is trying to solve?**
See https://jira.talendforge.org/browse/TDKN-219
 
**What is the chosen solution to this problem?**
See https://jira.talendforge.org/browse/TDKN-219
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDKN-219
 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/177/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/177/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/178/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/178/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/179/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/179/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/182/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/182/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/183/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/183/)

File: daikon-spring/daikon-spring-metrics/pom.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            <artifactId>spring-boot-starter-security</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>${commons-io.version}</version>
```
Comment (fabballe) [inline daikon-spring/daikon-spring-metrics/pom.xml]: why not handle version on dependencyManagement ?

File: daikon-spring/daikon-spring-metrics/src/main/resources/META-INF/spring.factories [hunk 2]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
+org.talend.daikon.spring.metrics.config.MetricsConfiguration
```
Comment (fabballe) [inline daikon-spring/daikon-spring-metrics/src/main/resources/META-INF/spring.factories]: why this file is on 2 line ?

File: daikon-spring/daikon-spring-metrics/src/test/java/org/talend/daikon/spring/metrics/io/SpanOutputStreamTest.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package org.talend.daikon.spring.metrics.io;
+
+public class SpanOutputStreamTest {
```
Comment (fabballe) [inline daikon-spring/daikon-spring-metrics/src/test/java/org/talend/daikon/spring/metrics/io/SpanOutputStreamTest.java]: still usefull ?

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/188/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/188/)

File: daikon-spring/daikon-spring-metrics/pom.xml (hunk 1 above)
Comment (fhuaulme) [reply daikon-spring/daikon-spring-metrics/pom.xml]: Yes, good idea. But I'd rather do this in a separate TDKN Jira.

File: daikon-spring/daikon-spring-metrics/src/main/resources/META-INF/spring.factories (hunk 2 above)
Comment (fhuaulme) [reply daikon-spring/daikon-spring-metrics/src/main/resources/META-INF/spring.factories]: For formatting purposes (seems to be longer than max line size).

File: daikon-spring/daikon-spring-metrics/pom.xml (hunk 1 above)
Comment (fhuaulme) [reply daikon-spring/daikon-spring-metrics/pom.xml]: https://jira.talendforge.org/browse/TDKN-223

File: daikon-spring/daikon-spring-metrics/src/test/java/org/talend/daikon/spring/metrics/io/SpanOutputStreamTest.java (hunk 3 above)
Comment (fhuaulme) [reply daikon-spring/daikon-spring-metrics/src/test/java/org/talend/daikon/spring/metrics/io/SpanOutputStreamTest.java]: No, it isn't. Good catch.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/189/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/189/)

**Pull request merged**